src/GafferImage/Noise.cpp

```diff
@@ -0,0 +1,343 @@
+{
+	ImageNode::hashDataWindow( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+Imath::Box2i Noise::computeDataWindow( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue().getDisplayWindow();
+}
+
+IECore::ConstCompoundDataPtr Noise::computeMetadata( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return outPlug()->metadataPlug()->defaultValue();
+}
+
+void Noise::hashChannelNames( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelNames( output, context, h );
+	layerPlug()->hash( h );
+}
+
+IECore::ConstStringVectorDataPtr Noise::computeChannelNames( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	std::string channelNamePrefix = layerPlug()->getValue();
+	if( !channelNamePrefix.empty() )
+	{
+		channelNamePrefix += ".";
+	}
+
+	StringVectorDataPtr resultData = new StringVectorData();
+	vector<string> &result = resultData->writable();
+
+	result.push_back( channelNamePrefix + "R" );
+	result.push_back( channelNamePrefix + "G" );
+	result.push_back( channelNamePrefix + "B" );
+	result.push_back( channelNamePrefix + "A" );
+
+	return resultData;
+}
+
+void Noise::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelData( output, context, h );
+
+	minOutputPlug()->hash( h );
+	maxOutputPlug()->hash( h );
+	sizePlug()->hash( h );
+	depthPlug()->hash( h );
+	octavesPlug()->hash( h );
+	gainPlug()->hash( h );
+	lacunarityPlug()->hash( h );
+	transformPlug()->hash( h );
+
+	V2i tileOrigin = context->get<V2i>( ImagePlug::tileOriginContextName );
+	h.append( tileOrigin );
+
+	string channelName = context->get<std::string>( ImagePlug::channelNameContextName );
+	h.append( channelName );
+}
+
+IECore::ConstFloatVectorDataPtr Noise::computeChannelData( const std::string &channelName, const Imath::V2i &tileOrigin, const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	const M33f transform = transformPlug()->matrix().inverse();
+
+	FloatVectorDataPtr resultData = new FloatVectorData;
+	vector<float> &result = resultData->writable();
+	result.reserve( ImagePlug::tileSize() * ImagePlug::tileSize() );
+	const V2f size = sizePlug()->getValue();
+	const float lacunarity = lacunarityPlug()->getValue();
+	const int octaves = octavesPlug()->getValue();
+	const float gain = gainPlug()->getValue();
+	const float depth = depthPlug()->getValue();
+
+	float n;
+	for( int y = 0; y < ImagePlug::tileSize(); ++y )
+	{
+		for( int x = 0; x < ImagePlug::tileSize(); ++x )
+		{
+			// screen space pixel coordinates
+			V2f p( tileOrigin.x + x, tileOrigin.y + y );
+			p *= transform;
+			n = fBm( p, octaves, V3f(size.x, size.y, 1 ), lacunarity, gain, depth );
+			result.push_back( remap( n, -.5, .5, minOutputPlug()->getValue(), maxOutputPlug()->getValue() ) );
```

Comment:
Hang on! This might be one cause of the speed difference - we should be calling `getValue()` once outside of this loop, not for every pixel...